### ISSUE TITLE
Clang format file related to priest's is doing something weird?

### ISSUE THREAD

**Issue by seanpeters86** [simulationcraft/simc#5266]
### Describe the bug
Both myself and @scamille have done this and I'm not sure exactly what causes it. Basically if you run clang-format on `sc_priest.hpp` it moves `#include "simulationcraft.hpp"` below the other include statements. However, if I do this simc now fails to build:

```
1>------ Build started: Project: simc_vs2019, Configuration: Release x64 ------
1>sc_priest.cpp
1>sc_priest_discipline.cpp
1>sc_priest_holy.cpp
1>sc_priest_shadow.cpp
1>B:\simc\engine\player\pet_spawner_impl.hpp(148,25): error C2027: use of undefined type 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest_shadow.cpp)
1>B:\simc\engine\player\runeforge_data.hpp(11): message : see declaration of 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest_shadow.cpp)
1>B:\simc\engine\player\pet_spawner_impl.hpp(152,22): error C2027: use of undefined type 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest_shadow.cpp)
1>B:\simc\engine\player\runeforge_data.hpp(11): message : see declaration of 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest_shadow.cpp)
1>B:\simc\engine\player\pet_spawner_impl.hpp(148,25): error C2027: use of undefined type 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest.cpp)
1>B:\simc\engine\player\runeforge_data.hpp(11): message : see declaration of 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest.cpp)
1>B:\simc\engine\player\pet_spawner_impl.hpp(152,22): error C2027: use of undefined type 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest.cpp)
1>B:\simc\engine\player\runeforge_data.hpp(11): message : see declaration of 'spell_data_t' (compiling source file ..\engine\class_modules\priest\sc_priest.cpp)
1>Done building project "simc_vs2019.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

### Expected behavior
I'd expect either the error to not happen when clang-format is ran, or clang-format to not alter this import statement location.

### To Reproduce
Set imports on `sc_priest.hpp` to:

```cpp
#pragma once
#include "player/covenant.hpp"
#include "player/pet_spawner.hpp"
#include "sc_enums.hpp"

#include "simulationcraft.hpp"
```

**Comment by scamille** [comment]
Since removing precompiled headers, this should in theory no longer be a problem. Let me double-check.
Edit: I think the problem was me not recompiling after pushing :-) And the priest.hpp header is just missing some includes.

**Comment by scamille** [comment]
Fixed in b15370ecb875a7ed510c3ac8f977141f77750a63 and bbdceb75a8fab0079f97a71ffe7ff4c37828d90e

While the reordering of simulationcraft.hpp was showing the problem, the underlying problem were missing includes in the pet spawner code.

And just to also note here: It is intentional that simulationcraft.hpp goes to the bottom now that precompiled headers are gone.
